### ISSUE TITLE
Incorrect upgrade button for Legacy / Multi-Cluster App 

### ISSUE THREAD

**Issue by richard-cox** [rancher/dashboard#4797]
**Describe the bug**
- Legacy apps installed pre-2.6.3 will show up in the Apps and Marketplace Apps list. 
- The upgrade button can show an incorrect version to upgrade to and the route the button takes the user to is incorrect (the new helm install process instead of the legacy app install

**Additional context**
- Work has already started on this in https://github.com/rancher/dashboard/pull/4795, though was very close to release so a safer/low bar solution of https://github.com/rancher/dashboard/pull/4796 was used for https://github.com/rancher/dashboard/issues/4786 - 

**Other comments**
- Any redirect to a legacy page should be gated on the legacy feature flag being enabled
- Need to confirm if multi-cluster apps will ever show up (they don't seem to at the moment)
- Need to confirm how the `catalog.cattle.io/rancher-version` annotation works

**Comment by nwmac** [comment]
@richard-cox This is tagged 2.6.4 - needed or can this be bumped?

**Comment by richard-cox** [comment]
@nwmac bumped!

**Comment by nwmac** [comment]
Won't fix due to deprecation
